website_field_autocomplete/controllers/main.py

```diff
@@ -17,15 +17,30 @@ class Website(Website):
         methods=['GET'],
         website=True,
     )
-    def _get_autocomplete_data(self, model, **kwargs):
-        res = []
+    def _get_field_autocomplete(self, model, **kwargs):
+        """ Return json autocomplete data """
         domain = json.loads(kwargs.get('domain', "[]"))
         fields = json.loads(kwargs.get('fields', "[]"))
         limit = kwargs.get('limit', None)
+        res = self._get_autocomplete_data(model, domain, fields, limit)
+        return json.dumps(res.values())
+
+    def _get_autocomplete_data(self, model, domain, fields, limit=None):
+        """ Gets and returns raw record data
+        Params:
+            model: Model name to query on
+            domain: Search domain
+            fields: List of fields to get
+            limit: Limit results to
+        Returns:
+            Dict of record dicts, keyed by ID
+        """
+        res = {}
         if limit:
             limit = int(limit)
-        for rec_id in request.env[model].search(domain, limit=limit):
-            res.append({
+        self.record_ids = request.env[model].search(domain, limit=limit)
+        for rec_id in self.record_ids:
+            res[rec_id.id] = {
```

Comment:
This was how I originally implemented it, until you made a comment originally about public access ;)

JsonRPC requires a non-public user session to operate. Downside of the implementation there was requiring a session, thus this would not work without logging in. It now does, as can be seen by the demo on `website.menu`